lib/mediawiki/query/lists/all.rb

```diff
@@ -0,0 +1,188 @@
+module MediaWiki
+  module Query
+    module Lists
+      module All
+        # Gets all categories on the entire wiki.
+        # @param limit [Int] The maximum number of categories to get. Defaults
+        #   to 500. Cannot be greater than 500 for normal users, or 5000 for
+        #   bots.
+        # @see https://www.mediawiki.org/wiki/API:Allcategories MediaWiki
+        #   Allcategories API Docs
+        # @since 0.7.0
+        # @return [Array] An array of all categories.
+        def get_all_categories(limit = 500)
+          params = {
+            action: 'query',
+            list: 'allcategories',
+            aclimit: get_limited(limit)
+          }
+
+          response = post(params)
+
+          ret = []
+          response['query']['allcategories'].each { |c| ret << c['*'] }
+
+          ret
+        end
+
+        # Gets all the images on the wiki.
+        # @param limit [Int] The maximum number of images to get. Defaults to
+        #   500. Cannot be greater than 500 for normal users, or 5000 for bots.
+        # @see https://www.mediawiki.org/wiki/API:Allimages MediaWiki Allimages
+        #   API Docs
+        # @since 0.7.0
+        # @return [Array] An array of all images.
+        def get_all_images(limit = 500)
+          params = {
+            action: 'query',
+            list: 'allimages',
+            ailimit: get_limited(limit)
+          }
+
+          response = post(params)
+
+          ret = []
+          response['query']['allimages'].each { |i| ret << i['name'] }
+
+          ret
+        end
+
+        # Gets all pages within a namespace integer.
+        # @param namespace [Int] The namespace ID.
+        # @param limit [Int] See #get_all_images
+        # @see https://www.mediawiki.org/wiki/API:Allpages MediaWiki Allpages
+        #   API Docs
+        # @since 0.8.0
+        # @return [Array] An array of all page titles.
+        def get_all_pages_in_namespace(namespace, limit = 500)
+          params = {
+            action: 'query',
+            list: 'allpages',
+            apnamespace: namespace,
+            aplimit: get_limited(limit)
+          }
+
+          response = post(params)
+
+          ret = []
+          response['query']['allpages'].each { |p| ret << p['title'] }
+
+          ret
+        end
+
+        # Gets all users, or all users in a group.
+        # @param group [String] The group to limit this query to.
+        # @param limit [Int] See #get_all_images.
+        # @see https://www.mediawiki.org/wiki/API:Allusers MediaWiki Allusers
+        #   API Docs
+        # @since 0.8.0
+        # @return [Hash] A hash of all users, names are keys, IDs are values.
+        def get_all_users(group = nil, limit = 500)
+          params = {
+            action: 'query',
+            list: 'allusers',
+            aulimit: get_limited(limit)
+          }
```

Comment:
Any reason why? That isn't the existing style used throughout the rest of the project.